Applications/Utils/ModelPreparation/PartitionMesh/PartitionMesh.cpp

```diff
@@ -0,0 +1,146 @@
+/*!
+  \file PartitionMesh.cpp
+  \date   2016.05
+
+  \brief  A tool for mesh partitioning.
+
+  \copyright
+  Copyright (c) 2012-2016, OpenGeoSys Community (http://www.opengeosys.org)
+             Distributed under a Modified BSD License.
+               See accompanying file LICENSE.txt or
+               http://www.opengeosys.org/project/license
+
+*/
+
+#include <tclap/CmdLine.h>
+
+#ifdef _MSC_VER
+#include <windows.h>
+#else
+#include <sys/time.h>
+#include <climits>
+#endif
+
+#include "Applications/ApplicationsLib/LogogSetup.h"
+#include "BaseLib/FileTools.h"
+#include "BaseLib/CPUTime.h"
+#include "BaseLib/RunTime.h"
+
+#include "MeshLib/IO/readMeshFromFile.h"
+
+#include "NodeWiseMeshPartitioner.h"
+
+int main(int argc, char* argv[])
+{
+    ApplicationsLib::LogogSetup logog_setup;
+
+    std::string m_str =
+        "Partition a mesh for parallel computing."
+        "The tasks of this tool are in twofold:\n"
+        "1. Convert mesh file to the input file of the partitioning tool,\n"
+        "2. Partition a mesh using the partitioning tool,\n"
+        "\tcreate the mesh data of each partition,\n"
+        "\trenumber the node indicies of each partition,\n"
+        "\tand output the results for parallel computing.";
+
+    TCLAP::CmdLine cmd(m_str, ' ', "0.1");
+    TCLAP::ValueArg<std::string> mesh_input(
+        "i", "mesh-input-file",
+        "the name of the file containing the input mesh", true, "",
+        "file name of input mesh");
+    cmd.add(mesh_input);
+    TCLAP::ValueArg<int> nparts("n", "np", "the number of partitions", false, 2,
+                                "integer");
+    cmd.add(nparts);
+
+    TCLAP::SwitchArg ogs2metis_flag(
+        "s", "ogs2metis",
+        "Indicator to convert the ogs mesh file to METIS input file", cmd,
+        false);
+
+    TCLAP::SwitchArg exe_metis_flag(
+        "m", "exe_metis", "Call mpmetis inside the programme via system().",
+        false);
+    cmd.add(exe_metis_flag);
+
+    TCLAP::SwitchArg lh_elems_flag(
+        "q", "lh_elements", "Mixed linear and high order elements.", false);
+    cmd.add(lh_elems_flag);
+
+    TCLAP::SwitchArg ascii_flag("a", "ascii", "Enable ASCII output.", false);
+    cmd.add(ascii_flag);
+
+    cmd.parse(argc, argv);
+
+    BaseLib::RunTime run_timer;
+    run_timer.start();
+    BaseLib::CPUTime CPU_timer;
+    CPU_timer.start();
+
+    const std::string ifile_name = mesh_input.getValue();
+    const std::string file_name_base = BaseLib::dropFileExtension(ifile_name);
+    std::unique_ptr<MeshLib::Mesh> mesh_ptr(
+        MeshLib::IO::readMeshFromFile(file_name_base + ".vtu"));
```

Comment:
In case both of full name and base name can be acceptable in the command input, e.g.
partmesh -s -i foo
or
partmesh -s -i foo.vtu